retriever/lib/socrata.py

```diff
@@ -0,0 +1,260 @@
+                })
+
+    except HTTPError as e:
+        print("HTTPError : ", e)
+        return []
+
+    return resources
+
+
+def find_socrata_dataset_by_id(dataset_id):
+    """Returns metadata for the following dataset id"""
+    url = SOCRATA_BASE_URL
+    params = {"ids": dataset_id}
+    try:
+        response = requests.get(
+            url,
+            params,
+            headers={
+                'user-agent':
+                    'Weecology/Data-Retriever Package Manager: http://www.data-retriever.org/'
+            },
+            allow_redirects=True)
+
+        if response.status_code == 400:
+            print(response.json()["error"])
+            return {}
+        elif response.status_code == 200:
+            resource = {}
+            data = response.json()
+
+            if len(data["results"]) != 0:
+                metadata = data["results"][0]
+
+                if (metadata["resource"]["lens_view_type"] !=
+                        "tabular") or (metadata["resource"]["type"] == "map"):
+                    resource["error"] = "Dataset cannot be downloaded using Socrata API"
+                    resource["datatype"] = [
+                        metadata["resource"]["type"],
+                        metadata["resource"]["lens_view_type"]
+                    ]
+
+                elif len(metadata["resource"]["columns_name"]) == 0:
+                    print("Cannot download the dataset using {}".format(dataset_id))
+                    print("Try downloading using its parent id : {}".format(
+                        metadata["resource"]["parent_fxf"][0]))
+                    return resource
+
+                else:
+                    resource["name"] = metadata["resource"]["name"]
+                    resource["id"] = metadata["resource"]["id"]
+                    resource["description"] = metadata["resource"]["description"]
+                    resource["datatype"] = metadata["resource"]["lens_view_type"]
+                    resource["keywords"] = list(
+                        set(metadata["classification"]["categories"] +
+                            metadata["classification"]["tags"])) + ["socrata"]
+                    resource["domain"] = metadata["metadata"]["domain"]
+                    resource["homepage"] = metadata["link"]
+            else:
+                print("No socrata dataset exists for the id : {dataset_id}".format(
+                    dataset_id=dataset_id))
+
+    except HTTPError as e:
+        print("HTTPError : ", e)
+        return {}
+
+    return resource
+
+
+def update_socrata_contents(json_file, script_name, url, resource):
+    """Update the contents of the json script"""
+    if "archived" in json_file.keys():
+        json_file.pop("archived")
+    if ("resources" in json_file.keys()) and (len(json_file["resources"]) != 0) and (
+            "path" in json_file["resources"][0].keys()):
+        json_file["resources"][0].pop("path")
```

Comment:
if all(["resources" in json_file.keys()), json_file["resources"], "path" in json_file["resources"][0].keys()])